astropy/table/table.py

```diff
@@ -913,13 +918,24 @@ def _convert_data_to_col(self, data, copy=True, default_name=None, dtype=None, n
             # does not happen.
             col_cls = self._get_col_cls_for_table(data)
 
-        elif self._is_mixin_for_table(data):
+        elif data_is_mixin:
             # Copy the mixin column attributes if they exist since the copy below
             # may not get this attribute.
             col = col_copy(data, copy_indices=self._init_indices) if copy else data
             col.info.name = name
             return col
 
+        elif data0_is_mixin:
+            # Handle case of a sequence of a mixin, e.g. [1*u.m, 2*u.m].
+            try:
+                col = data[0].__class__(data)
+                col.info.name = name
+                return col
+            except Exception:
+                # If that didn't work for some reason, just turn it into np.array of object
```

Comment:
Are we sure we want to default to turning the list into an object array? My sense would be to be a bit less forgiving in this case. E.g., the list may contain quantities with incompatible units or incompatible `TimeDelta` scales - both of which would be more likely mistakes. My sense would be to force users to create object arrays if they really want them.

(Over at numpy, there has been discussion - but no action - on never automatically creating object arrays; see https://github.com/numpy/numpy/issues/5353)

Reply:
I think that this is a somewhat significant API change that would need to be done in a graduated way starting with future deprecation warnings or whatnot.  I'm pretty sure some of our Chandra code would break, and I be we aren't the only ones.  Silly stuff like below can be handy sometimes:
```
t['stuff'] = [random_obj1, random_obj2]
```
Yes, there is a workaround but ...  At the very least, that API change should start as a separate issue with some focused discussion.  My feeling would be to follow whatever numpy does or does not do on that front.

Reply:
I thought it currently didn't work, but a quick test shows that I am wrong. Given that, you're right, no point breaking backward compatibility!